### PR TITLE
dbt dev docs updates - linting and some extra context/links

### PR DESCRIPTION
**Pull request by SorenSpicknall**
# Description

Following up on #2874 by bringing in changes that were originally drafted in #2889, including strict linting of the dbt development guidance markdown file and addition of some mild additional context + links.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

## How has this been tested?

Markdown linted successfully

## Post-merge follow-ups

- [x] No action required
- [ ] Actions required (specified below)

### CODE REVIEW

File: docs/warehouse/developing_dbt_models.md [hunk 1]
```diff
@@ -182,26 +200,35 @@ The first step is running your changes in the test/staging environment. You can
 What to test/check will vary based on what you're doing, but below are some example things to consider.
 
 #### Column values
+
 Are the values in your column/model what you expect? For example, are there nulls? Does the column have all the values you anticipated (for example, if you have a day of the week column, is data from all 7 days present)? If it's numeric, what are the minimum and maximum values; do they make sense (for example, if you have a percentage column, is it always between 0 and 100)? What is the most common value?
+
 * To check nulls:
-    ```
+
+    ```{sql}
```
Comment (lauriemerrell) [inline docs/warehouse/developing_dbt_models.md:208]: Looks like JupyterBook markdown doesn't know about SQL, can you revert these? I don't [see an equivalent here](https://jupyterbook.org/en/stable/content/content-blocks.html)
image

**Review by lauriemerrell — CHANGES_REQUESTED** [review]

File: docs/warehouse/developing_dbt_models.md (hunk 1 above)
Comment (SorenSpicknall) [reply docs/warehouse/developing_dbt_models.md:208]: That was just a labeling issue; I shouldn't have fenced them in brackets. Fixing in a new commit momentarily.

Comment (lauriemerrell) [reply docs/warehouse/developing_dbt_models.md:208]: Actually it looks like it might be `sql` without curly braces

Comment (lauriemerrell) [reply docs/warehouse/developing_dbt_models.md:208]: See https://raw.githubusercontent.com/cal-itp/data-infra/main/docs/analytics_tools/jupyterhub.md for reference

Comment (lauriemerrell) [reply docs/warehouse/developing_dbt_models.md:208]: Uses `python` and `bash`

Comment (lauriemerrell) [reply docs/warehouse/developing_dbt_models.md]: Ope sorry, my GH hadn't refreshed to show your response

File: docs/warehouse/developing_dbt_models.md [hunk 2]
```diff
@@ -80,7 +84,8 @@ tests_and_docs --> merge_model_changes
 ```
 
 (identify-bug)=
-### Identify the cause of your bug.
+
+### Identify the cause of your bug
```
Comment (lauriemerrell) [inline docs/warehouse/developing_dbt_models.md:88]: if you are deleting the periods in these section headers can you delete them in the Mermaid diagram too? goal was consistency. Not a huge deal but if we are doing this kind of editing may as well keep it consistent if we can

**Review by lauriemerrell — APPROVED** [review]

File: docs/warehouse/developing_dbt_models.md (hunk 1 above)
Comment (SorenSpicknall) [reply docs/warehouse/developing_dbt_models.md]: Resolved in [a6ac2ef](https://github.com/cal-itp/data-infra/pull/2896/commits/a6ac2ef85367618c81f49b0a45f13811ff4043b0)

File: docs/warehouse/developing_dbt_models.md (hunk 2 above)
Comment (SorenSpicknall) [reply docs/warehouse/developing_dbt_models.md:88]: Resolved in [e2ff4ca](https://github.com/cal-itp/data-infra/pull/2896/commits/e2ff4cafe8678fbd6bcfa2a82c4d9124d6b4cef2)

**Pull request merged**